src/common/src/types/to_binary.rs

```diff
@@ -13,39 +13,75 @@
 // limitations under the License.
 
 use bytes::{Bytes, BytesMut};
+use chrono::{TimeZone, Utc};
 use postgres_types::{ToSql, Type};
 
-use super::{DatumRef, ScalarRefImpl};
+use super::{DataType, DatumRef, ScalarRefImpl};
 use crate::error::Result;
 // Used to convert ScalarRef to text format
 pub trait ToBinary {
+    // to_text impl to_binary_with_type using default type of ScalarRefImpl.
```

Comment:
`to_binary`

Actually this is easier to remove compared to `to_text`. It is only used internally by `to_binary_with_type` right now. Any reason to keep it?

Reply:
Indeed, we can remove it.